src/qat/purr/compiler/execution.py

```diff
@@ -243,27 +244,30 @@ def validate(self, instructions: List[Instruction]):
                 )
             if isinstance(inst, (Pulse, CustomPulse)):
                 duration = inst.duration
-                if isinstance(duration, Number) and duration > MaxPulseLength:
-                    raise ValueError(
-                        f"Max Waveform width is {MaxPulseLength} s "
-                        f"given: {inst.duration} s"
-                    )
+                if isinstance(duration, Number):
+                    values = np.array([duration])
                 elif isinstance(duration, Variable):
-                    values = next(
-                        iter(
-                            [
-                                sw.variables[duration.name]
-                                for sw in instructions
-                                if isinstance(sw, Sweep)
-                                and duration.name in sw.variables.keys()
-                            ]
+                    values = np.array(
+                        next(
+                            iter(
+                                [
+                                    sw.variables[duration.name]
+                                    for sw in instructions
+                                    if isinstance(sw, Sweep)
+                                    and duration.name in sw.variables.keys()
+                                ]
+                            )
```

Comment:
too bulky, compute the associated sweep first and then get the variable value from it